src/test/java/thrift/logic/commands/AddIncomeCommandTest.java

```diff
@@ -48,20 +49,43 @@ public void execute_incomeAcceptedByModel_addSuccessful() {
 
     @Test
     public void undo_undoSuccessful() {
-        ModelStubUndoAddIncome modelStub = new ModelStubUndoAddIncome();
+        ModelStubUndoRedoAddIncome modelStub = new ModelStubUndoRedoAddIncome();
 
         Income validIncome = new IncomeBuilder().build();
+        modelStub.addIncome(validIncome);
+        AddIncomeCommand addIncomeCommand = new AddIncomeCommand(validIncome);
+        modelStub.keepTrackCommands(addIncomeCommand);
+        assertEquals(1, modelStub.getThrift().getTransactionList().size());
+        assertFalse(modelStub.undoableCommandStack.isEmpty());
+
+        Undoable undoable = modelStub.getPreviousUndoableCommand();
+        undoable.undo(modelStub);
+        assertEquals(0, modelStub.getThrift().getTransactionList().size());
+        assertTrue(modelStub.undoableCommandStack.isEmpty());
+    }
 
+    @Test
+    public void redo_redoSuccessful() {
```

Comment:
Is the method name supposed to be `undo_redoSuccessful` since it does undo followed by redo?

Reply:
I follows the JUnit test convention  `<whatIsBeingTested>_<descriptionOfTestInputs>_<expectedOutcome>`

Reply:
Oh, that is great then :smile: